Source/WebCore/platform/mediarecorder/cocoa/MediaRecorderPrivateWriterWebM.mm

```diff
@@ -535,14 +536,29 @@ static String codecStringForMediaVideoCodecId(FourCharCode codec)
 {
     m_lastTimestamp = MonotonicTime::now();
     Locker locker { m_dataLock };
-    m_data.append(data);
+    if (m_dataBuffer.size() + data.size() > m_dataBuffer.capacity())
+        flushDataBuffer();
+    if (data.size() < m_dataBuffer.capacity())
+        m_dataBuffer.append(data);
+    else
+        m_data.append(data);
+}
+
+void MediaRecorderPrivateWriterWebM::flushDataBuffer()
+{
+    assertIsHeld(m_dataLock);
+    if (m_dataBuffer.isEmpty())
+        return;
+    m_data.append(std::exchange(m_dataBuffer, { }));
+    m_dataBuffer.reserveInitialCapacity(s_dataBufferSize);
```

Comment:
If we were lazy allocating it, we would not need this line, we would do everything in appendData() and restart slim whenever taking data.